### PR TITLE
Completed Home Page

### PR DESCRIPTION
**Pull request by bhattabhi013**

### CODE REVIEW

**Comment by bhattabhi013** [comment]
@vishwanath1004  , please review

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 import { PrivateGuard } from './core/gurads/private.guard';
 
 const routes: Routes = [
   {
     path: '',
-    redirectTo: 'auth',
+    redirectTo: 'home',
     pathMatch: 'full'
   },
   {
```
Comment (vishwanath1004) [inline src/app/app-routing.module.ts:11]: Add guard for login also

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,3 +1,33 @@
 <ion-app>
-      <ion-router-outlet id="main-content"></ion-router-outlet>
+    <ion-menu side="start" contentId="main-content" type="overlay">
```
Comment (vishwanath1004) [inline src/app/app.component.html:2]: Create a new module as layout and place all the headers , side menu and footer component.
Create new component for side menu and use that component here
Create new component for Header and footer and that component here

File: src/app/components/pie-chart/pie-chart.component.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div>
+  <p class="report-text"> Project Report </p>
```
Comment (vishwanath1004) [inline src/app/components/pie-chart/pie-chart.component.html]: As this component is reusable so title should be dynamic.

File: src/app/components/pie-chart/pie-chart.component.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-pie-chart',
+  templateUrl: './pie-chart.component.html',
+  styleUrls: ['./pie-chart.component.scss'],
+})
+export class PieChartComponent  implements OnInit {
+  @Input() chartData: any;
+  chartLabels = ["Started", "Not Started"];
```
Comment (vishwanath1004) [inline src/app/components/pie-chart/pie-chart.component.ts]: Take these from constant files

File: src/app/components/project-details/project-details.component.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  selector: 'app-project-details',
+  templateUrl: './project-details.component.html',
+  styleUrls: ['./project-details.component.scss'],
+})
+export class ProjectDetailsComponent  implements OnInit {
+  @Input() projectName: string = "Project Name";
+  @Input() projectStatus: string | undefined;
+  @Input() taskCount: any = 1;
+
+  onCourseBtnClicked(projectName: any) {
```
Comment (vishwanath1004) [inline src/app/components/project-details/project-details.component.ts]: method and variable  names should be meaningful.

File: src/app/components/project-details/project-details.component.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<ion-card class="custom-card">
```
Comment (vishwanath1004) [inline src/app/components/project-details/project-details.component.html]: Make component as project-card

File: src/app/pages/home/home.page.html [hunk 7]
```diff
@@ -1,13 +1,60 @@
...
-
 <ion-content [fullscreen]="true">
-  <ion-header collapse="condense">
-    <ion-toolbar>
-      <ion-title size="large">home</ion-title>
-    </ion-toolbar>
-  </ion-header>
+  <div style="padding: 1%;">
+    <ion-grid>
+      <ion-row>
```
Comment (vishwanath1004) [inline src/app/pages/home/home.page.html]: Make separate component for header items.

File: src/app/pages/home/home.page.ts [hunk 8]
```diff
@@ -7,9 +11,62 @@ import { Component, OnInit } from '@angular/core';
...
+  chartData: any;
+  
+  async ionViewWillEnter(){
+    this.makeApiCall();
+  }
+  async makeApiCall() {
+    let data =  await this.localStorage.getLocalData(localKeys.USER_DETAILS)
+    const authToken = JSON.parse(data).access_token;
+    this.name = JSON.parse(data).user.name;
+    const headers = new HttpHeaders({
```
Comment (vishwanath1004) [inline src/app/pages/home/home.page.ts]: To get header part write one common method in api service file so we can easily resuable that.

File: src/app/pages/home/home.page.ts [hunk 9]
```diff
@@ -7,9 +11,62 @@ import { Component, OnInit } from '@angular/core';
...
+    ) { }
 
+  projects: any;
+  name: any;
+  started: any = 0;
+  notStarted: any = 0;
+  chartData: any;
+  
+  async ionViewWillEnter(){
+    this.makeApiCall();
```
Comment (vishwanath1004) [inline src/app/pages/home/home.page.ts]: use proper method names.

File: src/assets/i18n/en.json [hunk 10]
```diff
@@ -23,14 +23,28 @@
...
+        "DESC_LBL" : "Project Description",
+        "DESC_PLACEHOLDER" : "Enter project description",
+        "BTN" : "Create Project"
+    },
     "EMAIL" : "Email Id",
     "PASSWORD" : "Password",
     "EMAIL_PLACEHOLDER": "Enter your e-mail ID",
     "PASSWORD_PLACEHOLDER" : "Enter your password",
     "SIGN_IN" : "Sign in?",
     "data": {
```
Comment (vishwanath1004) [inline src/assets/i18n/en.json]: Every where you are using the Caps for all strings so please maintain consistency

File: src/app/pages/home/home.page.scss [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  #container strong {
+    font-size: 20px;
+    line-height: 26px;
+  }
+  
+  #container p {
+    font-size: 16px;
+    line-height: 22px;
+  
+    color: black;
```
Comment (vishwanath1004) [inline src/app/pages/home/home.page.scss]: Colors should come from varaibles file it help us to make theming

**Pull request merged**